build_tools/benchmarks/common/benchmark_driver.py

```diff
@@ -112,21 +80,50 @@ def run(self) -> None:
           model_name_filter=self.config.model_name_filter)
 
       for benchmark_case in benchmark_cases:
-        (benchmark_info, benchmark_results_filename,
-         capture_filename) = self.__get_benchmark_info_and_output_paths(
-             category, benchmark_case)
+        benchmark_info = self.__get_benchmark_info_from_case(
+            category=category, benchmark_case=benchmark_case)
+        benchmark_name = str(benchmark_info)
+
+        # Sanity check for the uniqueness of benchmark names.
+        if benchmark_name in self._seen_benchmark_names:
+          raise ValueError(
+              f"Found duplicate benchmark {benchmark_name} in the suites.")
+        self._seen_benchmark_names.add(benchmark_name)
+
+        results_path, capture_path = self.__get_output_paths(benchmark_name)
+        # If we continue from the previous results, check and skip if the result
+        # files exist.
+        if self.config.continue_from_previous:
+          if results_path is not None and results_path.exists():
+            self.finished_benchmarks.append((benchmark_info, results_path))
+            results_path = None
+
+          if capture_path is not None and capture_path.exists():
+            self.finished_captures.append(capture_path)
+            capture_path = None
 
         # Skip if no need to benchmark and capture.
-        if not benchmark_results_filename and not capture_filename:
+        if results_path is None and capture_path is None:
           continue
 
-        benchmark_key = str(benchmark_info)
-        print(f"--> Benchmark started: {benchmark_key} <--")
+        print(f"--> Benchmark started: {benchmark_name} <--")
 
         try:
-          self.run_benchmark_case(benchmark_case, benchmark_results_filename,
-                                  capture_filename)
+          self.run_benchmark_case(benchmark_case, results_path, capture_path)
         except Exception as e:
+          # Delete unfinished results if they exist.
+          # TODO(#11087): Use missing_ok=True once we move to Python 3.8.
+          if results_path is not None:
+            try:
+              results_path.unlink()
+            except FileNotFoundError:
+              pass
```

Comment:
Could use `is_file`?
```suggestion
          if results_path is not None and results_path.is_file():
            results_path.unlink()
```